serialization/journal.proto

```diff
@@ -155,8 +155,65 @@ message XY {
   required double y = 2;
 }
 
+message Interval {
+  required double min = 1;
+  required double max = 2;
+}
+
+message OrbitalElements {
+  required double sidereal_period = 1;
+  required double nodal_period = 2;
+  required double anomalistic_period = 3;
+  required double nodal_precession = 4;
+  required Interval mean_semimajor_axis = 5;
+  required Interval mean_eccentricity = 6;
+  required Interval mean_inclination = 7;
+  required Interval mean_longitude_of_ascending_nodes = 8;
+  required Interval mean_argument_of_periapsis = 9;
+}
+
+message OrbitRecurrence {
+  // νₒ.
+  required int32 nuo = 1;
+  // Dᴛₒ.
+  required int32 dto = 2;
+  // Cᴛₒ.
+  required int32 cto = 3;
+  required int32 number_of_revolutions = 4;
+  required double equatorial_shift = 5;
+  required double base_interval = 6;
+  required double grid_interval = 7;
+  required int32 subcycle = 8;
+}
+
+message EquatorialCrossings {
+  required Interval longitudes_reduced_to_ascending_pass = 1;
+  required Interval longitudes_reduced_to_descending_pass = 2;
+}
+
+message OrbitGroundTrack {
+  required EquatorialCrossings equatorial_crossings = 1;
+  // TODO(egg): mean solar times of the nodes.
+}
+
+message OrbitAnalysis {
+  required double progress_of_next_analysis = 1;
+  required int32 primary_index = 2;
+  required double mission_duration = 3;
+  // NOTE(egg): we currently have no support in the interface generator for
+  // optional non-string fields in interchange structs.  We use separate
```

Comment:
Why don't we fix the generator instead of doing this horrible hack?

Reply:
Fixing the generator is exceedingly complicated; we cannot have an `std::optional<T>` on the C++ side as it does not have standard layout, so we would need to define our own optional type. On the C# side, we would need to marshal the `T?` completely by hand.